json-path/src/main/java/com/jayway/jsonpath/Criteria.java

```diff
@@ -296,6 +296,56 @@ public Criteria subsetof(Collection<?> c) {
         return this;
     }
 
+    /**
+     * The <code>anyof</code> operator selects objects for which the specified field is
+     * an array that contain at least an element in the specified array.
+     *
+     * @param o the values to match against
+     * @return the criteria
+     */
+    public Criteria anyof(Object... o) {
+        return subsetof(Arrays.asList(o));
+    }
+
+    /**
+     * The <code>anyof</code> operator selects objects for which the specified field is
+     * an array that contain at least an element in the specified array.
+     *
+     * @param c the values to match against
+     * @return the criteria
+     */
+    public Criteria anyof(Collection<?> c) {
+        notNull(c, "collection can not be null");
+        this.criteriaType = RelationalOperator.ANYOF;
+        this.right = new ValueListNode(c);
+        return this;
+    }
+
+    /**
+     * The <code>noneof</code> operator selects objects for which the specified field is
+     * an array that does not contain any of the elements of the specified array.
+     *
+     * @param o the values to match against
+     * @return the criteria
+     */
+    public Criteria noneof(Object... o) {
+        return subsetof(Arrays.asList(o));
```

Comment:
This should be `noneof` instead of `subsetof` ?

Reply:
> This should be `noneof` instead of `subsetof` ?

Indeed you are right it should be `noneof`

Reply:
> > This should be `noneof` instead of `subsetof` ?
> 
> Indeed you are right it should be `noneof`

Sorry, I can't help you, never has this issue. Here is a quick example:

```java
  public static void main(String[] args) {
    String json = "{\n"
        + "    \"store\": {\n"
        + "        \"book\": [\n"
        + "            {\n"
        + "                \"categories\": [\"reference\"],\n"
        + "                \"author\": \"Nigel Rees\",\n"
        + "                \"title\": \"Sayings of the Century\",\n"
        + "                \"price\": 8.95\n"
        + "            },\n"
        + "            {\n"
        + "                \"categories\": [\"fiction\",\"war\"],\n"
        + "                \"author\": \"Evelyn Waugh\",\n"
        + "                \"title\": \"Sword of Honour\",\n"
        + "                \"price\": 12.99\n"
        + "            },\n"
        + "            {\n"
        + "                \"categories\": [\"fiction\"],\n"
        + "                \"author\": \"Herman Melville\",\n"
        + "                \"title\": \"Moby Dick\",\n"
        + "                \"isbn\": \"0-553-21311-3\",\n"
        + "                \"price\": 8.99\n"
        + "            },\n"
        + "            {\n"
        + "                \"categories\": [\"fiction\",\"fantasy\"],\n"
        + "                \"author\": \"J. R. R. Tolkien\",\n"
        + "                \"title\": \"The Lord of the Rings\",\n"
        + "                \"isbn\": \"0-395-19395-8\",\n"
        + "                \"price\": 22.99\n"
        + "            }\n"
        + "        ],\n"
        + "        \"bicycle\": {\n"
        + "            \"color\": \"red\",\n"
        + "            \"price\": 19.95\n"
        + "        }\n"
        + "    },\n"
        + "    \"expensive\": 10\n"
        + "}";


    JSONArray jsonArray = JsonPath.compile("$.store.book[?(@.categories SUBSETOF [\"fiction\",\"war\",\"fantasy\"])]").read(json);
    System.out.println(jsonArray.size()); // 3
    System.out.println(jsonArray.get(0)); // {categories=["fiction","war"], author=Evelyn Waugh, title=Sword of Honour, price=12.99}
    System.out.println(jsonArray.get(1)); // {categories=["fiction"], author=Herman Melville, title=Moby Dick, isbn=0-553-21311-3, price=8.99}
    System.out.println(jsonArray.get(2)); // {categories=["fiction","fantasy"], author=J. R. R. Tolkien, title=The Lord of the Rings, isbn=0-395-19395-8, price=22.99}

    jsonArray = JsonPath.compile("$.store.book[?(@.categories ANYOF [\"war\",\"fantasy\"])]").read(json);
    System.out.println(jsonArray.size()); // 2
    System.out.println(jsonArray.get(0)); // {categories=["fiction","war"], author=Evelyn Waugh, title=Sword of Honour, price=12.99}
    System.out.println(jsonArray.get(1)); // {categories=["fiction","fantasy"], author=J. R. R. Tolkien, title=The Lord of the Rings, isbn=0-395-19395-8, price=22.99}

    jsonArray = JsonPath.compile("$.store.book[?(@.categories NONEOF [\"war\",\"fantasy\"])]").read(json);
    System.out.println(jsonArray.size()); // 2
    System.out.println(jsonArray.get(0)); // {categories=["reference"], author=Nigel Rees, title=Sayings of the Century, price=8.95}
    System.out.println(jsonArray.get(1)); // {categories=["fiction"], author=Herman Melville, title=Moby Dick, isbn=0-553-21311-3, price=8.99}
  }
```